src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/Http/HttpRequestMessageTest.cs

```diff
@@ -346,6 +547,93 @@ public void ToString_DefaultAndNonDefaultInstance_DumpAllFields()
                 "}", rm.ToString());
         }
 
+        [Fact]
+        public void ToString_Blob_DefaultAndNonDefaultInstance_DumpAllFields()
+        {
+            var rm = new HttpRequestMessage();
+            string expected =
+                    "Method: GET, RequestUri: '<null>', Version: " +
+                    _expectedRequestMessageVersion.ToString(2) +
+                    $", Content: <null>, Headers:{Environment.NewLine}{{{Environment.NewLine}}}";
+            Assert.Equal(expected, rm.ToString());
+
+            rm.Method = HttpMethod.Put;
+            rm.RequestUri = new Uri("blob:http://a.com/");
+            rm.Version = new Version(1, 0);
+            rm.Content = new StringContent("content");
+
+            // Note that there is no Content-Length header: The reason is that the value for Content-Length header
+            // doesn't get set by StringContent..ctor, but only if someone actually accesses the ContentLength property.
+            Assert.Equal(
+                "Method: PUT, RequestUri: 'blob:http://a.com/', Version: 1.0, Content: " + typeof(StringContent).ToString() + ", Headers:" + Environment.NewLine +
+                $"{{{Environment.NewLine}" +
+                "  Content-Type: text/plain; charset=utf-8" + Environment.NewLine +
+                "}", rm.ToString());
+
+            rm.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/plain", 0.2));
+            rm.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/xml", 0.1));
+            rm.Headers.Add("Custom-Request-Header", "value1");
+            rm.Content.Headers.Add("Custom-Content-Header", "value2");
+
+            Assert.Equal(
+                "Method: PUT, RequestUri: 'blob:http://a.com/', Version: 1.0, Content: " + typeof(StringContent).ToString() + ", Headers:" + Environment.NewLine +
+                "{" + Environment.NewLine +
+                "  Accept: text/plain; q=0.2" + Environment.NewLine +
+                "  Accept: text/xml; q=0.1" + Environment.NewLine +
+                "  Custom-Request-Header: value1" + Environment.NewLine +
+                "  Content-Type: text/plain; charset=utf-8" + Environment.NewLine +
+                "  Custom-Content-Header: value2" + Environment.NewLine +
+                "}", rm.ToString());
+        }
+
+        [Fact]
+        public void BlobStringUri_Marshal_CorrectValues()
+        {
+            Runtime.InvokeJS(@"
+                function typedArrayToURL(typedArray, mimeType) {
+                    //return URL.createObjectURL(new Blob([typedArray.buffer], {type: mimeType}))
+                    return 'blob:https://mdn.mozillademos.org/ca45b575-6348-4d3e-908a-3dbf3d146ea7';
```

Comment:
When creating a url like this `URL.createObjectURL` it will always be different so the url will never be fixed.  Just so happens to be fixed here because I took the test from  https://developer.mozilla.org/en-US/docs/Web/API/Blob and the URL.createObjectURL does not exist in the JS engine.

Reply:
I completely missed the comment right above that line. Maybe add a comment to explain your rationale, for people like me!

Reply:
Comments added